### DISCUSSION TITLE
property.ProductVersion.Major (Minor,Build,Revision) are returning error instead of the version.

### DISCUSSION THREAD

**Discussion by 89magnum89** [wixtoolset/issues#8599] [Questions]
I want to have together with the name of the app, the Major.Minor.Build version instead of a full version that has the Revision inside also.
So I have this code:

```
<?define ProductVersion=!(bind.FileVersion.MyFilId_exe) ?> <!-- This resturns 1.0.0.0 -->
<?define FullProductVersion=!(bind.property.ProductVersion) ?> <!-- This resturns 1.0.0.0 -->
<?define MajorVersion=!(bind.property.ProductVersion.Major)?> <!-- This SHOULD return 1 -->
<?define MinorVersion=!(bind.property.ProductVersion.Minor) ?> <!-- This SHOULD return 0 -->
<?define BuildNumber=!(bind.property.ProductVersion.Build) ?> <!-- This SHOULD return 0 -->
<?define ShortProductVersion = $(var.MajorVersion).$(var.MinorVersion).$(var.BuildNumber) ?> <!-- This SHOULD combine them into 1.0.0 -->

<Package Name="$(var.ProductName) $(var.ShortProductVersion)"/> 
```

The package name should be a string like "My App Name 1.0.0", but instead, it's throwing an error:
`<!-- Exception ON BUILD: Error (active)	WIX0298	Unresolved bind-time variable My App Name !(bind.property.ProductVersion.Major).!(bind.property.ProductVersion.Minor).!(bind.property.ProductVersion.Build). -->`

Am I doing something wrong or the Major,Minor,Build,Revision properties from ProductVersion is not working?

I saw in the ResolveDelayedFieldsCommand that you are executing the property.ProductVersion.Major, but it's going to the error reporting I assume in case the key is missing. And the key is missing from the dictionary VariableCache and from DelayedFields IEnumerable also.

**Comment by Korzhukov** [comment]
[#4308](https://github.com/wixtoolset/issues/issues/4308/) has is some explanation, but no fix available.

**Reply by 89magnum89** [reply]
Yeah...the year is 2015 there
